nomad/fsm.go

```diff
@@ -296,7 +296,15 @@ func (n *nomadFSM) applyDeregisterNode(buf []byte, index uint64) interface{} {
 		panic(fmt.Errorf("failed to decode request: %v", err))
 	}
 
-	if err := n.state.DeleteNode(index, req.NodeID); err != nil {
+	// Messages pre 0.9.3 use a single NodeID
```

Comment:
Could we explicitly handle the case where both are set?

Reply:
yes! added that test

Reply:
I've done that by making it an error, that may not be correct?